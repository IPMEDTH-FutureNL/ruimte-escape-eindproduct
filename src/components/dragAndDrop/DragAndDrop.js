import React, { useState } from 'react'
import { useNavigate } from "react-router-dom";
import PopUp from '../PopUp';
import HintEscape from '../HintsEscape';
import Bubble from '../../sound/Bubble.wav';


//planeten
import aarde  from "../../img/planeten/aarde.png"
import jupiter  from "../../img/planeten/jupiter.png"
import mars  from "../../img/planeten/mars.png"
import mercurius  from "../../img/planeten/mercurius.png"
import neptunes  from "../../img/planeten/neptunes.png"
import saturnus  from "../../img/planeten/saturnus.png"
import uranus  from "../../img/planeten/uranus.png"
import venus  from "../../img/planeten/venus.png"
import sun from "../../img/planeten/sun.png"

//planeten benaming met lijntje
import aarde__line from "../../img/planeten/aarde__line.png"
import jupiter__line from "../../img/planeten/jupiter__line.png"
import mars__line from "../../img/planeten/mars__line.png"
import mercurius__line from "../../img/planeten/mercurius__line.png"
import neptunes__line from "../../img/planeten/neptunes__line.png"
import saturnus__line from "../../img/planeten/saturnus__line.png"
import uranus__line from "../../img/planeten/uranus__line.png"
import venus__line from "../../img/planeten/venus__line.png"

const DragAndDrop = () => {
  const [count, setCount] = useState(0);
  const [wrong, setWrong] = useState(0);
  const [text, setText] = useState("");

  const navigate = useNavigate();
    
  const goToPuzzle = () => {
      navigate("/succes");
  }


  const allowDrop = (e) => {
    e.preventDefault();
  }

  const drag = (e) => 
  {
    e.dataTransfer.setData("content",e.target.id);
  }

  const drop = (e) =>
  {
    e.preventDefault();
        let image =e.dataTransfer.getData("content");
    if (e.target.id === document.getElementById(image).getAttribute('data-div')){ 
        const audio = new Audio(Bubble);
        audio.play();
        //alert("ok");          
        e.target.appendChild(document.getElementById(image));
        let changeOpacity = document.getElementById(e.target.id);
        changeOpacity.style.opacity = "1";
        changeOpacity.style.backgroundColor = "transparent";
        setCount(count + 1);
        console.log("good tries " + count);

        if(count === 7){
          setTimeout(() => {
            let currentPoints = localStorage.getItem("punten");
            let pointsToInt = parseInt(currentPoints);

            switch(wrong){
              case 0:
                pointsToInt = pointsToInt + 40;
                localStorage.setItem("punten", pointsToInt);
                break;
              case 1:
                pointsToInt = pointsToInt + 35;
                localStorage.setItem("punten", pointsToInt);
                break;
              case 2:
                pointsToInt = pointsToInt + 30;
                localStorage.setItem("punten", pointsToInt);
                break;
              case 3:
                pointsToInt = pointsToInt + 25;
                localStorage.setItem("punten", pointsToInt);
                break;
              case 4:
                pointsToInt = pointsToInt + 20;
                localStorage.setItem("punten", pointsToInt);
                break;
              case 5:
                pointsToInt = pointsToInt + 15;
                localStorage.setItem("punten", pointsToInt);
                break;
              case 6:
                pointsToInt = pointsToInt + 10;
                localStorage.setItem("punten", pointsToInt);
                break;
              case 7:
                pointsToInt = pointsToInt + 5;
                localStorage.setItem("punten", pointsToInt);
                break;
              case 8:
                pointsToInt = pointsToInt + 0;
                localStorage.setItem("punten", pointsToInt);
                break;
            }
          
            goToPuzzle();
            console.log("how many wrong " + wrong );
          }, 200)
        }
    }
    else{
       setWrong(wrong + 1);
       setText("Dat was helaas niet goed");
       let changeColor = document.getElementById(e.target.id);
       changeColor.style.backgroundColor = "red";
       changeColor.style.opacity = "1";
      //  let openpopup = document.getElementById("popup");
      //  openpopup.style.display = "block";
       console.log("bad tries " + wrong);
    }
  }


    return (
      <section className="mapPuzzle">
        <HintEscape type='dragAndDropHints'/>
        <img className="sun" alt="sun" src={sun}></img>
          <div className="board">
            <div className="board__piece">
                <div className="board__piece-square" id="div1" onDrop={(event) => drop(event)} onDragOver={(event) => allowDrop(event)} ></div>
                <img alt="planeten naam drag and drop" className="board__piece-name" src={mercurius__line}  draggable="false"/>
            </div>
            <div className="board__piece">
                <div className="board__piece-square" id="div2" onDrop={(event) => drop(event)} onDragOver={(event) => allowDrop(event)} ></div>
                <img alt="planeten naam drag and drop" className="board__piece-name" src={venus__line}  draggable="false"/>
            </div>
            <div className="board__piece">
                <div className="board__piece-square" id="div3" onDrop={(event) => drop(event)} onDragOver={(event) => allowDrop(event)} ></div>
                <img alt="planeten naam drag and drop" className="board__piece-name" src={aarde__line}  draggable="false"/>
            </div>
            <div className="board__piece">
                <div className="board__piece-square" id="div4" onDrop={(event) => drop(event)} onDragOver={(event) => allowDrop(event)} ></div>
                <img alt="planeten naam drag and drop" className="board__piece-name" src={mars__line}  draggable="false"/>
            </div>
            <div className="board__piece">
                <div className="board__piece-square" id="div5" onDrop={(event) => drop(event)} onDragOver={(event) => allowDrop(event)} ></div>
                <img alt="planeten naam drag and drop" className="board__piece-name" src={jupiter__line}  draggable="false"/>
            </div>
            <div className="board__piece">
                <div className="board__piece-square" id="div6" onDrop={(event) => drop(event)} onDragOver={(event) => allowDrop(event)} ></div>
                <img alt="planeten naam drag and drop" className="board__piece-name" src={saturnus__line}  draggable="false"/>
            </div>
            <div className="board__piece">
                <div className="board__piece-square" id="div7" onDrop={(event) => drop(event)} onDragOver={(event) => allowDrop(event)} ></div>
                <img alt="planeten naam drag and drop" className="board__piece-name" src={uranus__line}   draggable="false"/> 
            </div>
            <div className="board__piece">
                <div className="board__piece-square" id="div8" onDrop={(event) => drop(event)} onDragOver={(event) => allowDrop(event)} ></div>
                <img alt="planeten naam drag and drop" className="board__piece-name" src={neptunes__line} draggable="false" />
            </div>
          </div>

          <div className="puzzelstukjes">
            <img alt="planeten drag and drop" className="planet" id="drag3" data-div="div3" src={aarde} draggable="true" onDragStart={(event) => drag(event)}/>
            <img alt="planeten drag and drop" className="planet" id="drag5" data-div="div5" src={jupiter} draggable="true" onDragStart={(event) => drag(event)}/>
            <img alt="planeten drag and drop" className="planet" id="drag4"  data-div="div4" src={mars} draggable="true" onDragStart={(event) => drag(event)}/>
            <img alt="planeten drag and drop" className="planet" id="drag1"  data-div="div1" src={mercurius} draggable="true" onDragStart={(event) => drag(event)}/>
            <img alt="planeten drag and drop" className="planet" id="drag8"  data-div="div8" src={neptunes} draggable="true" onDragStart={(event) => drag(event)}/>
            <img alt="planeten drag and drop" className="planet" id="drag6"  data-div="div6" src={saturnus} draggable="true" onDragStart={(event) => drag(event)}/>
            <img alt="planeten drag and drop" className="planet" id="drag7"  data-div="div7" src={uranus} draggable="true" onDragStart={(event) => drag(event)}/>
            <img alt="planeten drag and drop" className="planet" id="drag2"  data-div="div2" src={venus} draggable="true" onDragStart={(event) => drag(event)}/>
          </div>

          <PopUp text={text}></PopUp>
      </section>
     
    )
}

export default DragAndDrop;