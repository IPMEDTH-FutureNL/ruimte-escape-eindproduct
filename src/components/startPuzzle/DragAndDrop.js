import React, { Component } from 'react'


//planeten
import aarde  from "../../img/planeten/aarde.png"
import jupiter  from "../../img/planeten/jupiter.png"
import mars  from "../../img/planeten/mars.png"
import mercurius  from "../../img/planeten/mercurius.png"
import neptunes  from "../../img/planeten/neptunes.png"
import saturnus  from "../../img/planeten/saturnus.png"
import uranus  from "../../img/planeten/uranus.png"
import venus  from "../../img/planeten/venus.png"

//planeten benaming met lijntje
import aarde__line from "../../img/planeten/aarde__line.png"
import jupiter__line from "../../img/planeten/jupiter__line.png"
import mars__line from "../../img/planeten/mars__line.png"
import mercurius__line from "../../img/planeten/mercurius__line.png"
import neptunes__line from "../../img/planeten/neptunes__line.png"
import saturnus__line from "../../img/planeten/saturnus__line.png"
import uranus__line from "../../img/planeten/uranus__line.png"
import venus__line from "../../img/planeten/venus__line.png"

export class DragAndDrop extends Component {
  constructor(props){
    super(props);
    this.state = {
      count : 0,
    }
}

allowDrop = (e) => {
  e.preventDefault();
}

drag = (e) => 
{
  e.dataTransfer.setData("content",e.target.id);
}

drop = (e) =>
{
  e.preventDefault();
        var image =e.dataTransfer.getData("content");
  if (e.target.id == document.getElementById(image).getAttribute('data-div')){ 
      //alert("ok");          
      e.target.appendChild(document.getElementById(image));
      let changeOpacity = document.getElementById(e.target.id);
      changeOpacity.style.opacity = "1";
      changeOpacity.style.backgroundColor = "transparent";
      this.setState({count : this.state.count + 1})
      console.log(this.state.count);

      if(this.state.count === 5){
        let showButton = document.getElementById("buttonOlifanten");
        showButton.style.display = "block";
      }
  }
  else{
     alert("wrong");
  }
}


  render() {
    return (
      <section className="mapPuzzle">
          <div className="board">
            <div className="board__piece">
                <div className="board__piece-square" id="div1" onDrop={(event) => this.drop(event)} onDragOver={(event) => this.allowDrop(event)} ></div>
                <img className="board__piece-name" src={mercurius__line} />
            </div>
            <div className="board__piece">
                <div className="board__piece-square" id="div2" onDrop={(event) => this.drop(event)} onDragOver={(event) => this.allowDrop(event)} ></div>
                <img className="board__piece-name" src={venus__line} />
            </div>
            <div className="board__piece">
                <div className="board__piece-square" id="div3" onDrop={(event) => this.drop(event)} onDragOver={(event) => this.allowDrop(event)} ></div>
                <img className="board__piece-name" src={aarde__line} />
            </div>
            <div className="board__piece">
                <div className="board__piece-square" id="div4" onDrop={(event) => this.drop(event)} onDragOver={(event) => this.allowDrop(event)} ></div>
                <img className="board__piece-name" src={mars__line}/>
            </div>
            <div className="board__piece">
                <div className="board__piece-square" id="div5" onDrop={(event) => this.drop(event)} onDragOver={(event) => this.allowDrop(event)} ></div>
                <img className="board__piece-name" src={jupiter__line} />
            </div>
            <div className="board__piece">
                <div className="board__piece-square" id="div6" onDrop={(event) => this.drop(event)} onDragOver={(event) => this.allowDrop(event)} ></div>
                <img className="board__piece-name" src={saturnus__line} />
            </div>
            <div className="board__piece">
                <div className="board__piece-square" id="div7" onDrop={(event) => this.drop(event)} onDragOver={(event) => this.allowDrop(event)} ></div>
                <img className="board__piece-name" src={uranus__line} /> 
            </div>
            <div className="board__piece">
                <div className="board__piece-square" id="div8" onDrop={(event) => this.drop(event)} onDragOver={(event) => this.allowDrop(event)} ></div>
                <img className="board__piece-name" src={neptunes__line} />
            </div>
          </div>

          <div className="puzzelstukjes">
            <img className="planet" id="drag3" data-div="div3" src={aarde} draggable="true" onDragStart={(event) => this.drag(event)}/>
            <img className="planet" id="drag5" data-div="div5" src={jupiter} draggable="true" onDragStart={(event) => this.drag(event)}/>
            <img className="planet" id="drag4"  data-div="div4" src={mars} draggable="true" onDragStart={(event) => this.drag(event)}/>
            <img className="planet" id="drag1"  data-div="div1" src={mercurius} draggable="true" onDragStart={(event) => this.drag(event)}/>
            <img className="planet" id="drag8"  data-div="div8" src={neptunes} draggable="true" onDragStart={(event) => this.drag(event)}/>
            <img className="planet" id="drag6"  data-div="div6" src={saturnus} draggable="true" onDragStart={(event) => this.drag(event)}/>
            <img className="planet" id="drag7"  data-div="div7" src={uranus} draggable="true" onDragStart={(event) => this.drag(event)}/>
            <img className="planet" id="drag2"  data-div="div2" src={venus} draggable="true" onDragStart={(event) => this.drag(event)}/>
          </div>
      </section>
     
    )
  }
}

export default DragAndDrop;