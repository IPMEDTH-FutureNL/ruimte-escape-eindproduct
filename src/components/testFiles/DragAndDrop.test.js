import renderer from 'react-test-renderer';
import DragAndDrop from "../dragAndDrop/DragAndDrop";
import {BrowserRouter as Router} from 'react-router-dom';

it('Test if Drag and drop game component is rendered', () => {
    const component = renderer.create(
        <Router>
            <DragAndDrop></DragAndDrop>,
        </Router>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})