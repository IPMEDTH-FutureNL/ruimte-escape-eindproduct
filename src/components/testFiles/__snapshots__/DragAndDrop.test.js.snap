// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if Drag and drop game component is rendered 1`] = `
Array [
  <section
    className="mapPuzzle"
  >
    <button
      aria-describedby="popup-1"
      className="hint-button"
      onClick={[Function]}
    >
       hints
    </button>
    <div
      className="punten"
    >
      Punten: 
    </div>
    <div
      className="board"
    >
      <div
        className="board__piece"
      >
        <div
          className="board__piece-square"
          id="div1"
          onDragOver={[Function]}
          onDrop={[Function]}
        />
        <img
          alt="planeten naam drag and drop"
          className="board__piece-name"
          src="mercurius__line.png"
        />
      </div>
      <div
        className="board__piece"
      >
        <div
          className="board__piece-square"
          id="div2"
          onDragOver={[Function]}
          onDrop={[Function]}
        />
        <img
          alt="planeten naam drag and drop"
          className="board__piece-name"
          src="venus__line.png"
        />
      </div>
      <div
        className="board__piece"
      >
        <div
          className="board__piece-square"
          id="div3"
          onDragOver={[Function]}
          onDrop={[Function]}
        />
        <img
          alt="planeten naam drag and drop"
          className="board__piece-name"
          src="aarde__line.png"
        />
      </div>
      <div
        className="board__piece"
      >
        <div
          className="board__piece-square"
          id="div4"
          onDragOver={[Function]}
          onDrop={[Function]}
        />
        <img
          alt="planeten naam drag and drop"
          className="board__piece-name"
          src="mars__line.png"
        />
      </div>
      <div
        className="board__piece"
      >
        <div
          className="board__piece-square"
          id="div5"
          onDragOver={[Function]}
          onDrop={[Function]}
        />
        <img
          alt="planeten naam drag and drop"
          className="board__piece-name"
          src="jupiter__line.png"
        />
      </div>
      <div
        className="board__piece"
      >
        <div
          className="board__piece-square"
          id="div6"
          onDragOver={[Function]}
          onDrop={[Function]}
        />
        <img
          alt="planeten naam drag and drop"
          className="board__piece-name"
          src="saturnus__line.png"
        />
      </div>
      <div
        className="board__piece"
      >
        <div
          className="board__piece-square"
          id="div7"
          onDragOver={[Function]}
          onDrop={[Function]}
        />
        <img
          alt="planeten naam drag and drop"
          className="board__piece-name"
          src="uranus__line.png"
        />
      </div>
      <div
        className="board__piece"
      >
        <div
          className="board__piece-square"
          id="div8"
          onDragOver={[Function]}
          onDrop={[Function]}
        />
        <img
          alt="planeten naam drag and drop"
          className="board__piece-name"
          src="neptunes__line.png"
        />
      </div>
    </div>
    <div
      className="puzzelstukjes"
    >
      <img
        alt="planeten drag and drop"
        className="planet"
        data-div="div3"
        draggable="true"
        id="drag3"
        onDragStart={[Function]}
        src="aarde.png"
      />
      <img
        alt="planeten drag and drop"
        className="planet"
        data-div="div5"
        draggable="true"
        id="drag5"
        onDragStart={[Function]}
        src="jupiter.png"
      />
      <img
        alt="planeten drag and drop"
        className="planet"
        data-div="div4"
        draggable="true"
        id="drag4"
        onDragStart={[Function]}
        src="mars.png"
      />
      <img
        alt="planeten drag and drop"
        className="planet"
        data-div="div1"
        draggable="true"
        id="drag1"
        onDragStart={[Function]}
        src="mercurius.png"
      />
      <img
        alt="planeten drag and drop"
        className="planet"
        data-div="div8"
        draggable="true"
        id="drag8"
        onDragStart={[Function]}
        src="neptunes.png"
      />
      <img
        alt="planeten drag and drop"
        className="planet"
        data-div="div6"
        draggable="true"
        id="drag6"
        onDragStart={[Function]}
        src="saturnus.png"
      />
      <img
        alt="planeten drag and drop"
        className="planet"
        data-div="div7"
        draggable="true"
        id="drag7"
        onDragStart={[Function]}
        src="uranus.png"
      />
      <img
        alt="planeten drag and drop"
        className="planet"
        data-div="div2"
        draggable="true"
        id="drag2"
        onDragStart={[Function]}
        src="venus.png"
      />
    </div>
    <div
      className="pop-up"
      id="popup"
    >
      <p
        className="pop-up-text"
      >
         
         
      </p>
      <button
        className="button primary"
        onClick={[Function]}
      >
        Verder!
      </button>
    </div>
  </section>,
  ",",
]
`;
